.github/workflows/source_aws.yml

```diff
@@ -98,20 +100,21 @@ jobs:
       - name: Checkout
         if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
         uses: actions/checkout@v3
-      - uses: actions/cache@v3
-        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-1.19.5-release-cache-${{ hashFiles('plugins/source/aws/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-1.19.5-release-cache-plugins-source-aws
       - name: Set up Go
         if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
+        id:   set-up-go
         uses: actions/setup-go@v3
         with:
           go-version-file: plugins/source/aws/go.mod
+          cache: false
+      - uses: actions/cache@v3
+        if: startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
+        with:
+          path: |
+                ~/.cache/go-build
+                ~/go/pkg/mod
+          key:          ${{ runner.os }}-go-1.19.5-release-cache-${{ hashFiles('plugins/source/aws/go.sum') }}
+          restore-keys: ${{ runner.os }}-go-1.19.5-release-cache-plugins-source-aws
```

Comment:
This should be addressed separately, as the go ver should be taken from setup-go output.
Additionally, we might want to change the naming convention there as well